core/src/main/java/eu/fasten/core/dependents/DependencyGraphBuilder.java

```diff
@@ -0,0 +1,195 @@
+
+    public Map<Revision, List<Dependency>> getDependencyList(DSLContext dbContext) {
+        return dbContext.select(PackageVersions.PACKAGE_VERSIONS.ID,
+                Packages.PACKAGES.PACKAGE_NAME,
+                PackageVersions.PACKAGE_VERSIONS.VERSION,
+                Dependencies.DEPENDENCIES.METADATA,
+                PackageVersions.PACKAGE_VERSIONS.CREATED_AT)
+                .from(Packages.PACKAGES)
+                .rightJoin(PackageVersions.PACKAGE_VERSIONS)
+                .on(PackageVersions.PACKAGE_VERSIONS.PACKAGE_ID.eq(Packages.PACKAGES.ID))
+                .leftJoin(Dependencies.DEPENDENCIES)
+                .on(Dependencies.DEPENDENCIES.PACKAGE_VERSION_ID.eq(PackageVersions.PACKAGE_VERSIONS.ID))
+                .and(PackageVersions.PACKAGE_VERSIONS.CREATED_AT.isNotNull())
+                .fetch()
+                .parallelStream()
+                .map(x -> {
+
+                    if (x.component4() != null) {
+                        return new AbstractMap.SimpleEntry<>(new Revision(x.component1(), x.component2(),
+                                x.component3().replaceAll("[\\n\\t ]", ""),
+                                x.component5()), Dependency.fromJSON(new JSONObject(x.component4().data())));
+                    } else {
+                        return new AbstractMap.SimpleEntry<>(new Revision(x.component1(), x.component2(),
+                                x.component3().replaceAll("[\\n\\t ]", ""),
+                                x.component5()), Dependency.empty);
+                    }
+                }).filter(Objects::nonNull)
+                .collect(Collectors.toConcurrentMap(
+                        AbstractMap.SimpleEntry::getKey,
+                        x -> List.of(x.getValue()),
+                        (x, y) -> {
+                            var z = new ArrayList<Dependency>();
+                            z.addAll(x);
+                            z.addAll(y);
+                            return z;
+                        }
+                ));
+    }
+
+    public List<Revision> findMatchingRevisions(List<Revision> revisions,
+                                                List<Dependency.VersionConstraint> constraints) {
+        if (revisions == null) {
+            return Collections.emptyList();
+        }
+        return revisions.stream().filter(r -> {
+            for (var constraint : constraints) {
+                if ((constraint.toString().startsWith("[") || constraint.toString().startsWith("("))
+                        && (constraint.toString().endsWith("]") || constraint.toString().endsWith(")"))) {
+                    if (checkVersionLowerBound(constraint, r.version) &&
+                            checkVersionUpperBound(constraint, r.version)) {
+                        return true;
+                    }
+                } else {
+                    if (constraint.lowerBound.equals(constraint.upperBound) &&
```

Comment:
Hello @amottier, sorry for the late response. The purpose of the findMatchingRevisions method is to find revisions that match  the version constraints of each dependency. The checks regarding a valid Version constraint are performed in the Dependency.VersionConstraint Class. Each Version Constraint has a lower & upper Bound attributes and two isLowerHardRequirement & isUpperHardRequirement binary values. This method https://github.com/fasten-project/fasten/blob/0c1b112713e0b05c788957920f1c9962b19a3464/core/src/main/java/eu/fasten/core/dependents/data/Dependency.java#L239-L264 ensures that always either the Version Constraint will include parenthesis or brackets or it will be a specific version constraint. So the case where a constraint neither  does include parenthesis or brackets nor has a lower bound equal to upper bound will not happen.